databricks_blueprints/helpers.py

```diff
@@ -0,0 +1,90 @@
+
+
+class DatabricksClient(object):
+    "Databricks connection object for use in running queries"
+    def __init__(self, token, instance_id):
+        self.token = token
+        self.base_url = f"https://{instance_id}/api/2.0"
+    
+    def get_headers(self):
+        return {'Authorization': f"Bearer {self.token}"}
+    
+    def get(self, endpoint, params={}):
+        api_headers = self.get_headers()
+        endpoint_url = self.base_url + endpoint
+        response = requests.get(endpoint_url, 
+                                headers=api_headers,
+                                params=params)
+        if response.status_code == 401: # invalid account token
+            sys.exit(errors.EXIT_CODE_INVALID_CREDENTIALS)
+        elif response.status_code == 404: # wrong instance_id
+            sys.exit(errors.EXIT_CODE_INVALID_INSTANCE)  
+        return response
+    
+    def post(self, endpoint, data={}):
+        api_headers = self.get_headers()
+        endpoint_url = self.base_url + endpoint
+        response = requests.post(endpoint_url, 
+                                headers=api_headers,
+                                data=data)
+        if response.status_code == 401: # invalid account token
+            sys.exit(errors.EXIT_CODE_INVALID_CREDENTIALS)
+        elif response.status_code == 404: # wrong instance_id
+            sys.exit(errors.EXIT_CODE_INVALID_INSTANCE)  
+        return response
+
+    def stream(self, endpoint, json={}):
+        api_headers = self.get_headers()
+        endpoint_url = self.base_url + endpoint
+        response = requests.post(endpoint_url, 
+                                headers=api_headers,
+                                json=json,
+                                stream=True)
+        if response.status_code == 401: # invalid account token
+            sys.exit(errors.EXIT_CODE_INVALID_CREDENTIALS)
+        elif response.status_code == 404: # wrong instance_id
+            sys.exit(errors.EXIT_CODE_INVALID_INSTANCE)
+        return response
+
+
+def start_cluster(token, instance_id, cluster_id):
+    """ 
+    Starts a Databricks Cluster and saves the status in the artifacts folder. 
+    Ideally this function should be run first whenever someone tries to 
+    interact with Databricks using our CLI applications.
+    
+    see: https://docs.databricks.com/dev-tools/api/latest/clusters.html#start
+    """
+    databricks_client = DatabricksClient(token, instance_id)
+    start_cluster_endpoint = "/clusters/start"
+    payload = {"cluster_id": cluster_id}
+    start_response = databricks_client.post(start_cluster_endpoint, data=payload)
+    if start_response.status_code == requests.codes.ok:
+        print(f"Cluster: {cluster_id} started successfully")
+    elif start_response.status_code == 400: # Cluster in RESTARTING state
+        print(f"Cannot start Cluster: {cluster_id}. Cluster is currently RESTARTING")
+        sys.exit(errors.EXIT_CODE_CLUSTER_STATUS_RESTARTING)
```

Comment:
When running this function, it is always returning `{"error_code":"MALFORMED_REQUEST","message":"Invalid JSON given in the body of the request - failed to parse given JSON"}`, resulting in a 400, which results in us exiting with a `RESTARTING` error code.

1) Need to verify why the error message is occuring.
2) Need to make sure we're not relying on the 400 for just restarting. Parse the response to specifically extract the specific issues. Then, once we've exhausted all known messages, we return with an UKNNOWN_ERROR code.